File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0, right = nums.size()-1;
        vector<int> ans(nums.size(), 0);
        int k = nums.size()-1;
        while (left <= right){
            int l = nums[left];
            int r = nums[right];

            if (l*l < r*r){
                ans[k] = r*r;
                k-=1;
                right-=1;
            } else {
                ans[k] = l*l;
                k-=1;
                left+=1;
            }
        }

        return ans;
    }
};

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.go
func sortedSquares(nums []int) []int {
    i, j, k := 0, len(nums)-1, len(nums)-1
    square := make([]int, len(nums))

    for i <= j {
        x := (nums[i]*nums[i])
        y := (nums[j]*nums[j])

        if x >= y {
            square[k] = x
            i++
        } else {
            square[k] = y
            j--
        }
        k--
    }

    return square
}